### PR TITLE
Move Instructed PL Sections to My PL page

### PR DESCRIPTION
**Pull request by TurnerRiley**
Moves the Professional Learning sections the user is teaching from studio.code.org/home to [studio.code.org/my-professional-learning](https://studio.code.org/my-professional-learning).

Following the spec, this PR should:
- Show the table of the PL sections the user is teaching
   - Show it only for facilitators (under the "My Facilitator Center" tab), universal instructors (under the "Instructors" tab), and peer reviewers (under the "Instructors" tab)
   - Show the "Create Section" component
- Show the co-teacher invites banners for professional learning sections that were on the Teacher Dashboard
- Remove the PL sections table from the Teacher Dashboard and replace with a short message with a redirect to the My PL page

### Note about recreating the Owned Sections Table
Unfortunately, there are a lot of different table setups used across the Teacher Dashboard and the My PL page so it was not as simple as moving the `OwnedSectionsTable` component over because the styling was pretty off as the My PL page's styles, the imported styles from `tableConstants.js`, and the in-file styles within a `styles` object were all blending together. I figured taking the time to refactor `OwnedSectionsTable` to maintain its current styles on the Teacher Dashboard but have different functionality and styling on the My PL page would be scope creep. So, my solution was to make a similar `OwnedPlSectionsTable` that is the same content copied over but with anything unnecessary removed. I also consolidated the style updates for this in the `tableConstants.js` file so that it wasn't as confusing to track. If this seems like a concern I'm open to pushback and/or can make a ticket for building out this table following [the way Kelby did](https://github.com/code-dot-org/code-dot-org/pull/57811) with the `SelfPacedProgressTable` so that the styles can be better reused.

### View of page by permission
Facilitator:
(note: this Facilitator also has a non-PL section to demonstrate that this table only shows PL sections (both active and archived))
Facilitator

Universal instructor:
universalinstructor

Peer reviewer:
peer_reviewer

Teacher (should not be able to see it):
Teacherview

### Co-teacher invite
coteacherinvite

### Teacher Dashboard message where table used to be
teacherdashboard

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1426)
Spec: [here](https://docs.google.com/document/d/1Ee02kw_ei2x6wOS06K768dC-kUhGDlydjoE6BZipbPE/edit)
Figma: [here](https://www.figma.com/file/sZ6d1uhJWFksJ6j3Fnmbnc/PL-Flow-Improvements?type=design&node-id=1958-11140&mode=design&t=ZtBxwBfqS6Pg2PGB-0)

## Testing story
Local testing and adding a Storybook for `OwnedPlSectionsTable`.
OwnedPlSections

### CODE REVIEW

**Comment by dmcavoy** [comment]
I'm so excited to see this coming too life! Yay!!

Nit: Can we update the tab names to what is on this page in figma: https://www.figma.com/file/sZ6d1uhJWFksJ6j3Fnmbnc/PL-Flow-Improvements?type=design&node-id=1958-8045&mode=design&t=x4CDcEPgC2GtTFe6-0

Professional Learning
Facilitator Center
Regional Partner Center
Workshop Organizer Center
Instructor Center

**Comment by TurnerRiley** [comment]
> I'm so excited to see this coming too life! Yay!!
> 
> Nit: Can we update the tab names to what is on this page in figma: https://www.figma.com/file/sZ6d1uhJWFksJ6j3Fnmbnc/PL-Flow-Improvements?type=design&node-id=1958-8045&mode=design&t=x4CDcEPgC2GtTFe6-0
> 
> Professional Learning Facilitator Center Regional Partner Center Workshop Organizer Center Instructor Center

Oh gotcha I was going off the spec tab names, I'll update these! Do we still want to do "Instructors" instead of "Instructor Center" like we discussed offline?

File: apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx [hunk 1]
```diff
@@ -12,55 +13,90 @@ import SelfPacedProgressTable from './SelfPacedProgressTable';
...
+
+  // {
+  //   value: 'myRPCenter',
+  //   text: i18n.plLandingTabRPCenter(),
+  // },
+  // {
+  //   value: 'myWorkshopOrganizerCenter',
+  //   text: i18n.plLandingTabWorkshopOrganizerCenter(),
+  // },
+  // {
```
Comment (bethanyaconnor) [inline apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx]: nit: should this one still be commented out?

File: apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx [hunk 2]
```diff
@@ -142,50 +177,76 @@ export default function LandingPage({
...
+              </section>
+            )}
+            <section>
+              <Heading2>{i18n.plLandingRecommendedHeading()}</Heading2>
+              {RenderStaticRecommendedPL()}
+            </section>
+          </>
         )}
-        {deeperLearningCourseData?.length >= 1 && (
+        {['myFacilitatorCenter', 'instructors'].includes(currentTab) && (
```
Comment (bethanyaconnor) [inline apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx:210]: The facilitator tab should have more components at some point. While adding those components is outside the scope of this PR, it might be worth separating out this logic

File: apps/src/templates/teacherDashboard/OwnedSections.jsx [hunk 3]
```diff
@@ -65,12 +66,32 @@ class OwnedSections extends React.Component {
...
 
+  ownedSectionsTable = showHidden => {
+    const {isPlSections, sectionIds, hiddenSectionIds} = this.props;
+    const sectionsToShow = showHidden
+      ? hiddenSectionIds
+      : _.without(sectionIds, ...hiddenSectionIds);
+
+    return isPlSections ? (
+      <OwnedPlSectionsTable
+        isPlSections={true}
```
Comment (bethanyaconnor) [inline apps/src/templates/teacherDashboard/OwnedSections.jsx]: is `isPlSections` needed?

File: apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx]: I think since we haven't set up any content yet for those tabs I've been leaving them commented out until they are useful. Otherwise Regional Partners and Workshop Organizers will have empty tabs

File: apps/src/templates/teacherDashboard/OwnedSections.jsx (hunk 3 above)
Comment (TurnerRiley) [reply apps/src/templates/teacherDashboard/OwnedSections.jsx]: No not anymore, I guess I just wasn't sure how much I should change in the scope of this PR for updating `OwnedSections` and `OwnedSectionsTable`. Since this change would remove the need for `isPlSections` in several files would it be okay if I did a followup cleanup task for it?

File: apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx (hunk 2 above)
Comment (TurnerRiley) [reply apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx:210]: Yes for sure, my plan once we started implementing more components was to simplify the logic to a series of `{shouldRenderXComponent && (<XComponent/>)}` where shouldRenderXComponent would contain the logic of checking if the user has the right permissions and any other qualifications

File: apps/src/templates/teacherDashboard/OwnedSections.jsx (hunk 3 above)
Comment (TurnerRiley) [reply apps/src/templates/teacherDashboard/OwnedSections.jsx]: Cleanup work set up in this PR: https://github.com/code-dot-org/code-dot-org/pull/58075

Comment (bethanyaconnor) [reply apps/src/templates/teacherDashboard/OwnedSections.jsx]: But it's not a prop on `OwnedPlSectionsTable` is it?

Comment (bethanyaconnor) [reply apps/src/templates/teacherDashboard/OwnedSections.jsx]: Maybe to clarify, totally fine with cleaning it up on `OwnedSectionsTable` as a follow-up but I'd prefer it be removed here in this PR.

File: apps/test/unit/code-studio/pd/professional_learning_landing/landingPageTest.js
Comment (bethanyaconnor) [inline apps/test/unit/code-studio/pd/professional_learning_landing/landingPageTest.js:1]: Thank you for adding all of these tests!!

File: dashboard/test/ui/features/teacher_tools/pl_sections.feature [hunk 4]
```diff
@@ -1,45 +1,16 @@
 @no_mobile
 Feature: Professional learning Sections
 
-  Scenario: Create new professional learning section as levelbuilder
```
Comment (bethanyaconnor) [inline dashboard/test/ui/features/teacher_tools/pl_sections.feature:4]: Just so I'm clear, it looks like the test for "as a levelbuilder" was removed but creating PL sections is still covered by other tests in this file?

File: apps/src/templates/teacherDashboard/OwnedSections.jsx (hunk 3 above)
Comment (TurnerRiley) [reply apps/src/templates/teacherDashboard/OwnedSections.jsx]: Oh gotcha! Yes you're super right, just committed a fix!

File: dashboard/test/ui/features/teacher_tools/pl_sections.feature (hunk 4 above)
Comment (TurnerRiley) [reply dashboard/test/ui/features/teacher_tools/pl_sections.feature:4]: Yep! I removed it because this section was moved to the My PL page and levelbuilders won't have their own view to see this section there

**Review by bethanyaconnor — APPROVED** [review]
🎉
